enterprise/cmd/frontend/internal/batches/httpapi/file_handler.go

```diff
@@ -0,0 +1,235 @@
+package httpapi
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/opentracing/opentracing-go/log"
+	sglog "github.com/sourcegraph/log"
+
+	"github.com/sourcegraph/sourcegraph/enterprise/internal/batches/store"
+	btypes "github.com/sourcegraph/sourcegraph/enterprise/internal/batches/types"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/errcode"
+	"github.com/sourcegraph/sourcegraph/internal/observation"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+// FileHandler handles retrieving and uploading of files.
+type FileHandler struct {
+	logger     sglog.Logger
+	db         database.DB
+	store      BatchesStore
+	operations *Operations
+}
+
+type BatchesStore interface {
+	CountBatchSpecWorkspaceFiles(context.Context, store.ListBatchSpecWorkspaceFileOpts) (int, error)
+	GetBatchSpec(context.Context, store.GetBatchSpecOpts) (*btypes.BatchSpec, error)
+	GetBatchSpecWorkspaceFile(context.Context, store.GetBatchSpecWorkspaceFileOpts) (*btypes.BatchSpecWorkspaceFile, error)
+	UpsertBatchSpecWorkspaceFile(context.Context, *btypes.BatchSpecWorkspaceFile) error
+}
+
+// NewFileHandler creates a new FileHandler.
+func NewFileHandler(db database.DB, store BatchesStore, operations *Operations) http.Handler {
+	handler := &FileHandler{
+		logger:     sglog.Scoped("FileHandler", ""),
```

Comment:
It seems like we omit the description when creating scoped loggers about a third of the time ([113 instances with a description](https://sourcegraph.sourcegraph.com/search?q=context%3Asourcegraph+r%3A%5Egithub.com%2Fsourcegraph%2Fsourcegraph%24+%3A%5B%7E%5Cw*log%5D.Scoped%28...%2C+%3A%5B%7E%22.%2B%22%5D%29&patternType=structural&groupBy=repo), [52 without](https://sourcegraph.sourcegraph.com/search?q=context%3Asourcegraph+r%3A%5Egithub.com%2Fsourcegraph%2Fsourcegraph%24+%3A%5B%7E%5Cw*log%5D.Scoped%28...%2C+%22%22%29&patternType=structural&groupBy=repo)), so I don't think this is a code quality requirement for us, but we might want to attach a logger description here so there's no confusion with any other "file handlers":

```suggestion
		logger:     sglog.Scoped("FileHandler", "Batch Changes mounted file REST API handler"),
```